### PR TITLE
[Docs:  Getting Started / Block Development Environment] Updates on the text to improve clarity and preciseness

### PR DESCRIPTION
**Pull request by juanmaguitar**
## What?
Minor changes to some texts of the "Getting Started / Block Development Environment" section to improve clarity and preciseness

## Why?
To avoid confusion and have a more coherent narrative that helps developers understand and process this information faster.

### CODE REVIEW

File: docs/getting-started/devenv/README.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ The list goes on. While modern JavaScript development can be challenging, WordPr
 
 **The recommended Node.js version for block development is [Active LTS](https://nodejs.dev/en/about/releases/) (Long Term Support)**. However, there are times when you  need to to use different versions. A Node.js version manager tool like `nvm` is strongly recommended and allows you to easily change your `node` version when required. You will also need Node Package Manager (`npm`) and the Node Package eXecute (`npx`) to work with some WordPress packages. Both are installed automatically with Node.js.
 
-To install Node.js development tools and learn more about each, refer to the links below.
+To be able to use the [Node.js tools and packages provided by WordPress](https://github.com/WordPress/gutenberg/tree/trunk/packages) for block development you'll need to set a proper Node.js runtime environment in your machine. To learn more about how to do this, refer to the links below.
```
Comment (ndiego) [inline docs/getting-started/devenv/README.md]: > To be able to use the Node.js tools and [packages provided by WordPress](https://github.com/WordPress/gutenberg/tree/trunk/packages) for block development, you'll need to set a proper Node.js runtime environment on your machine.

File: docs/getting-started/devenv/nodejs-development-tools.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ The Node Package eXecute ([`npx`](https://docs.npmjs.com/cli/v10/commands/npx))
 
 It's recommended that you use [Node Version Manager](https://github.com/nvm-sh/nvm) (`nvm`) to install Node.js. This allows you to install and manage specific versions of `node`, which are installed locally in your home directory, avoiding any global permission issues.
 
-Here are the quick instructions for installing `node` using `nvm`. See the [complete installation guide](https://github.com/nvm-sh/nvm#installing-and-updating) for more details.
+Here are the quick instructions for installing `node` using `nvm` and set the recommended Node version for block development. See the [complete installation guide](https://github.com/nvm-sh/nvm#installing-and-updating) for more details.
```
Comment (ndiego) [inline docs/getting-started/devenv/nodejs-development-tools.md]: > Here are the quick instructions for installing `node` using `nvm` and setting the recommended Node.js version for block development.

**Review by ndiego — APPROVED** [review]
Looks good. I added a few grammatical tweaks.

**Comment by juanmaguitar** [comment]
I have added more changes to this PR that include:
- [this](https://github.com/WordPress/gutenberg/pull/54563#discussion_r1328969650) and [this](https://github.com/WordPress/gutenberg/pull/54563#discussion_r1328970027) feedback incorporated 
- Changed "Node.js development tools" section title to "Node development environment" 
- Changed `nodejs-development-tools.md` filename to `node-development-environment.md`
- Updated `manifest.json` and `toc.json` to reflect filename changes
- Fixed links pointing to each Node install sections 

@ndiego let me know your thoughts

**Review by ndiego — APPROVED** [review]
Looks good! 🚢

**Pull request merged**